sw/device/lib/dif/README.md

```diff
@@ -50,22 +50,41 @@ memory-mapped hardware, and modifying processor CSRs.
   DIFs must not call DIFs in other DIF libraries.
 * DIF library headers must be polyglot headers for C and C++.
 
-### DIF Guidance
+### DIF API Guidance
 
-The following rules must be followed by public DIF functions (those declared in
-the DIF library's header file). Internal DIF functions (those declared `static`
-and not declared in the DIF library's header file) should follow these rules but
-there are some relaxations of these rules for them described at the end.
+The following rules specify the basic API that each DIF must conform to. These
+rules specify the names of types, constants, and functions that each DIF must
+define for providing certain kinds of non-device-specific functionality (such as
+initializing handles or managing interrupts).
```

Comment:
It's probably worth including a reference to the template file in this document?